go-controller/pkg/ovn/master.go

```diff
@@ -712,6 +714,23 @@ func (oc *Controller) ensureNodeLogicalNetwork(nodeName string, hostSubnets []*n
 		}
 	}
 
+	// add the management IPs for this node to host network namespace's address set
+	hns := config.Kubernetes.HostNetworkNamespace
+	if hns != "" {
+		nsInfo, err := oc.waitForNamespaceLocked(hns)
+		if err != nil {
+			klog.Errorf("Failed to wait for namespace %s event (%v)",
+				hns, err)
+			return err
+		}
+		defer nsInfo.Unlock()
```

Comment:
might be better to put this whole thing into its own function, so that you're not holding onto the nsInfo lock for an unpredictable length of time